core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -792,11 +793,17 @@ public synchronized void startQueuedRebalance() {
 
       boolean updateTopology = false;
       boolean rebalance = false;
+      boolean updateStableTopology = false;
       if (rebalanceType == RebalanceType.NONE) {
          updateTopology = true;
       } else if (balancedCH.equals(currentCH)) {
          log.tracef("The balanced CH is the same as the current CH, not rebalancing");
          updateTopology = cacheTopology.getPendingCH() != null;
+         // After a cluster view change that leaves only 1 node, we don't need either a topology update or a rebalance
+         // but we must still update the stable topology
+         updateStableTopology =
```

Comment:
Wondering if it makes sense to have methods on CacheTopology that encapsulate some of the logic (I haven't checked about repeated uses).
e.g cacheTopology.needsStableTopologyUpdate(stableTopology)

Reply:
I'd like to avoid adding methods to `CacheTopology` that the application shouldn't care about, because it's exposed in `DistributionManager` and that makes changes harder.

For 9.3 I'd actually like to move internals like `ClusterCacheStatus` and `ClusterTopologyManagerImpl` away from using `CacheTopology`, so I can deprecate currentCH/pendingCH/unionCH/persistentUUIDs. The internal-only replacement would be a good location for such methods, though.